### ISSUE TITLE
Single quote in environment name breaks pre/post deploy Powershell scripts

### ISSUE THREAD

**Issue by Dalmirog** [OctopusDeploy/Issues#1710]
Environment Name: `Developmen't`
Pre-deploy script code: `"Some code"`

The deployment breaks when trying to resolve the path of the `CustomScriptConvention.*Deploy.ps1` script

```
At C:\Octopus\Applications\Prod\Developmen't\Testapp\1.0.0\Bootstrap.23571810-a
Error    16:48:59
172-4154-a491-7dd9d14beb2e.ps1:388 char:96
Error    16:48:59
+ ... Prod\Developmen't\Testapp\1.0.0\CustomScriptConvention.PreDeploy.ps1'
Error    16:48:59
+                                                                         ~
Error    16:48:59
The string is missing the terminator: '.
Error    16:48:59
    + CategoryInfo          : ParserError: (:) [], ParseException
Error    16:48:59
    + FullyQualifiedErrorId : TerminatorExpectedAtEndOfString
```

- The rest of the deployment seems to go just fine.
- Including multiple single quotes on the Environment name generates all sorts of (bad) behaviors.

Source: http://help.octopusdeploy.com/discussions/problems/35956

**Comment by mattbrooks2010** [comment]
This issue might need to be re-opened. In v3.0.5 it is fixed in the file

`Bootstrap.Octopus.Features.IISWebSite_BeforePostDeploy.ps1`

but not in the file

`Octopus.Features.WindowsService_BeforePostDeploy.ps1`

due to error

````
09:04:30   Info     |       The name_with_spaces_and_single_quote service already exists. It will be stopped and reconfigured.
09:04:30   Info     |       Stopping the name_with_spaces_and_single_quote service
09:04:30   Info     |       sc.exe config name_with_spaces_and_single_quote binPath= "\"path_with_spaces_and_single_quote\"" depend= / start= auto
09:04:30   Info     |       [SC] ChangeServiceConfig SUCCESS
09:04:30   Error    |       Get-WmiObject : Invalid query
09:04:30   Error    |       At path_with_spaces_and_single_quote\Octopus.Features.WindowsService_BeforePostDeploy.ps1:125 char:24
09:04:30   Error    |       + $status = Get-WMIObject <<<<  win32_service -filter ("name='" + $serviceName
09:04:30   Error    |       + "'") -computer "." | select -expand startMode
09:04:30   Error    |       + CategoryInfo          : InvalidOperation: (:) [Get-WmiObject], ManagementException
09:04:30   Error    |       + FullyQualifiedErrorId : GetWMIManagementException,Microsoft.PowerShell.Commands.GetWmiObjectCommand
09:04:30   Error    |       Script 'path_with_spaces_and_single_quote\Octopus.Features.WindowsService_BeforePostDeploy.ps1' returned non-zero exit code: 1
````

**Comment by michaelnoonan** [comment]
Hi @mattbrooks2010, thanks for the heads-up!

You're right and I fixed this as part of #1785 which was released as part of `3.0.7` but neglected to comment on this particular issue! I went through the PowerShell scripts for IIS and Windows Services and made sure everything is quoted and escaped correctly and tested each scenario. Hopefully we've covered everything now! (Famous last words I'm sure)

http://octopusdeploy.com/downloads/3.0.7

Hope that helps!
Mike
